test/extended/apibinding/apibinding_utils.go

```diff
@@ -0,0 +1,98 @@
+package apibinding
+
+import (
+	"fmt"
+
+	o "github.com/onsi/gomega"
+
+	exutil "github.com/kcp-dev/kcp-tests/test/extended/util"
+)
+
+// APIBinding struct defination
+type APIBinding struct {
+	APIVersion string `json:"apiVersion"`
+	Kind       string `json:"kind"`
+	Metadata   struct {
+		Name string `json:"name"`
+	} `json:"metadata"`
+	Spec struct {
+		Reference struct {
+			Workspace struct {
+				Path       string `json:"path"`
+				ExportName string `json:"exportName"`
+			} `json:"workspace"`
+		} `json:"reference"`
+	} `json:"spec"`
+}
+
+// ABOption uses function option mode to change the default values of APIBinding attributes
+type ABOption func(*APIBinding)
+
+// SetAPIBindingName replaces the default value of APIBinding name
```

Comment:
You can't replace a metadata.name. `sets` is the actual behavior.

Reply:
Fixed.